tasks/main.yml

```diff
@@ -64,12 +64,19 @@
 - name: Send config
   template:
     src: "config.py"
-    dest: "{{ tiaas_dir }}/config/local_settings.py"
+    dest: "{{ tiaas_config_dir }}/local_settings.py"
     owner: "{{ tiaas_user }}"
     group: "{{ tiaas_group }}"
     mode: 0750
   notify: 'reload tiaas'
 
+- name: Create symlink to Galaxy's stylesheet
+  file:
+    src: "{{ tiaas_galaxy_stylesheet }}"
+    path: "{{ tiaas_code_dir }}/training/static/training/style/galaxy.css"
+    state: link
+    force: yes
```

Comment:
I think this should trigger collecting the static files. So add a `register: __tiaas_symlink_css` or so, and then ensure the collectstatic runs on it too.

This could be a handler but I don't know how to ensure handler ordering (it needs to happen before the service is restarted.)

The other option is putting it as a pre-task in the systemd unit like openstack horizon does, but I really don't like the poorly visibility of that.

Reply:
Ahhh I missed the title of the PR, `run collectstatic unconditionally`. Yea, I'd still prefer register / when combination. The intention is to not say something has changed when it hasn't, because that makes me worry as an admin.

We should also add a third condition, that the galaxy client is rebuilt (using the internal variable for that), that we rebuild this task too.

We could probably replace the symlink with a copy action, which I think would be smarter about the source != dest.

Reply:
There are (at least) 3 cases when `collectstatic` should run:

1. tiaas2 master branch has been updated
2. Galaxy client rebuilt, so base.css may have changed
3. Files in the root `static` directory on the host have been (accidentally?) overwritten and need to be restored

To run `collectstatic` conditionally:

- Case 1: handled by the current role (`when: __tiaas_git_update_result is changed`)
- Case 2: can be handled by adding `register __foo` to the "copy galaxy stylesheet" task and `when: __foo is changed` to the "collect static files" task
- Case 3? I can't see an obvious easy solution

If we run `collectstatic` unconditionally:
- all cases are covered
- files are copied *"only if they have a modified timestamp greater than the timestamp of the file in STATIC_ROOT"* [(as per Django docs - see 3rd paragraph)](https://docs.djangoproject.com/en/3.1/ref/contrib/staticfiles/#django-admin-collectstatic)
- it is simple

For the sake of handling Case 3, I'd suggest leaving it unconditional; unless there's a solution to case 3. 
And that case, I think, is a reasonable expectation: if I run the playbook, the expected state on the host, including all files, should be restored. That said, it's very possible I'm not seeing an alternative solution.

Reply:
So for me, as a very heavy ansible user, something being marked as changed even when nothing changes, is a significant sign of code smell (and I'm not [alone](https://blog.evanstoner.net/post/120987608912/five-ansible-code-smells)). I'd like to avoid this if at all possible.

So. Checking the ansible module for managing django, what they do is run it every time and report changed only when it outputs a non-zero number of files updated.

```
  django_manage:
    command: collectstatic
```

so that as it is is fine! What a nice surprise. You've removed the `when`, so, that looks good then.

Reply:
> So for me, as a very heavy ansible user, something being marked as changed even when nothing changes, is a significant sign of code smell (and I'm not [alone](https://blog.evanstoner.net/post/120987608912/five-ansible-code-smells)). I'd like to avoid this if at all possible.

Indeed! I'm afraid I didn't think of it from the standpoint of ansible *reporting* the state as `changed` vs. `ok` (So yes, "change ignorance". Now I will think of it from that standpoint - thanks!). 
So, I guess this is good to go? (i hope i'm not missing anything)